worker/include/DepLibUring.hpp

```diff
@@ -0,0 +1,98 @@
+#ifndef MS_DEP_LIBURING_HPP
+#define MS_DEP_LIBURING_HPP
+
+#ifdef MS_LIBURING_SUPPORTED
+
+#include "DepLibUV.hpp"
+#include <functional>
+#include <liburing.h>
+#include <queue>
+
+class DepLibUring
+{
+public:
+	using onSendCallback = const std::function<void(bool sent)>;
+
+	/* Struct for the user data field of SQE and CQE. */
+	struct UserData
+	{
+		uint8_t store[1500]{};
+		onSendCallback* cb{ nullptr };
+		size_t idx{ 0 };
+	};
+
+	/* Number of submission queue entries (SQE). */
+	static constexpr size_t QueueDepth{ 1024 * 4 };
+
+	static bool IsSupported();
+	static void ClassInit();
+	static void ClassDestroy();
+	static void StartPollingCQEs();
+	static void StopPollingCQEs();
+	static bool PrepareSend(
+	  int sockfd, const void* data, size_t len, const struct sockaddr* addr, onSendCallback* cb);
+	static bool PrepareWrite(
+	  int sockfd, const void* data1, size_t len1, const void* data2, size_t len2, onSendCallback* cb);
+	static void Submit();
+	static void SetActive();
+	static bool IsActive();
+
+	class LibUring;
+
+	thread_local static LibUring* liburing;
+
+public:
+	class LibUring
+	{
+	public:
+		LibUring();
+		~LibUring();
+		void StartPollingCQEs();
+		void StopPollingCQEs();
+		bool PrepareSend(
+		  int sockfd, const void* data, size_t len, const struct sockaddr* addr, onSendCallback* cb);
+		bool PrepareWrite(
+		  int sockfd, const void* data1, size_t len1, const void* data2, size_t len2, onSendCallback* cb);
+		void Submit();
+		void SetActive()
+		{
+			this->active = true;
+		}
+		bool IsActive() const
+		{
+			return this->active;
+		}
+		io_uring* GetRing()
+		{
+			return std::addressof(this->ring);
+		}
+		int GetEventFd() const
+		{
+			return this->efd;
+		}
+		void ReleaseUserDataEntry(size_t idx)
+		{
+			this->availableUserDataEntries.push(idx);
+		}
+
+	private:
+		UserData* GetUserData();
+
+	private:
+		// io_uring instance.
+		io_uring ring;
+		// Event file descriptor to watch for completions.
+		int efd;
+		// libuv handle used to poll uring completions.
+		uv_poll_t* uvHandle{ nullptr };
+		// Whether we are currently sending RTP over io_uring.
+		bool active{ false };
+		// Pre-allocated UserData entries.
+		UserData userDataBuffer[QueueDepth]{};
+		// Indexes of available UserData entries.
+		std::queue<size_t> availableUserDataEntries;
+	};
+};
+
+#endif
+#endif
```

Comment:
So let's not check the macro here. Instead, the rest of the code should NOT include `DepLibUring.hpp` if the macro is not given. Consistency with your own proposal:
- We only include `DepLibUring.hpp` if the macro is given.
- We only call methods in `DepLibUring` if the macro is given.
- Everywhere.
- We don't add any macro check in `DepLibUring.hpp` or `DepLibUring.cpp` since those files should ONLY be included and built if the macro is given.

```suggestion
#ifndef MS_DEP_LIBURING_HPP
#define MS_DEP_LIBURING_HPP

#include "DepLibUV.hpp"
#include <functional>
#include <liburing.h>
#include <queue>

class DepLibUring
{
public:
	using onSendCallback = const std::function<void(bool sent)>;

	/* Struct for the user data field of SQE and CQE. */
	struct UserData
	{
		uint8_t store[1500]{};
		onSendCallback* cb{ nullptr };
		size_t idx{ 0 };
	};

	/* Number of submission queue entries (SQE). */
	static constexpr size_t QueueDepth{ 1024 * 4 };

	static bool IsSupported();
	static void ClassInit();
	static void ClassDestroy();
	static void StartPollingCQEs();
	static void StopPollingCQEs();
	static bool PrepareSend(
	  int sockfd, const void* data, size_t len, const struct sockaddr* addr, onSendCallback* cb);
	static bool PrepareWrite(
	  int sockfd, const void* data1, size_t len1, const void* data2, size_t len2, onSendCallback* cb);
	static void Submit();
	static void SetActive();
	static bool IsActive();

	class LibUring;

	thread_local static LibUring* liburing;

public:
	class LibUring
	{
	public:
		LibUring();
		~LibUring();
		void StartPollingCQEs();
		void StopPollingCQEs();
		bool PrepareSend(
		  int sockfd, const void* data, size_t len, const struct sockaddr* addr, onSendCallback* cb);
		bool PrepareWrite(
		  int sockfd, const void* data1, size_t len1, const void* data2, size_t len2, onSendCallback* cb);
		void Submit();
		void SetActive()
		{
			this->active = true;
		}
		bool IsActive() const
		{
			return this->active;
		}
		io_uring* GetRing()
		{
			return std::addressof(this->ring);
		}
		int GetEventFd() const
		{
			return this->efd;
		}
		void ReleaseUserDataEntry(size_t idx)
		{
			this->availableUserDataEntries.push(idx);
		}

	private:
		UserData* GetUserData();

	private:
		// io_uring instance.
		io_uring ring;
		// Event file descriptor to watch for completions.
		int efd;
		// libuv handle used to poll uring completions.
		uv_poll_t* uvHandle{ nullptr };
		// Whether we are currently sending RTP over io_uring.
		bool active{ false };
		// Pre-allocated UserData entries.
		UserData userDataBuffer[QueueDepth]{};
		// Indexes of available UserData entries.
		std::queue<size_t> availableUserDataEntries;
	};
};

#endif
```